### PR TITLE
`@Publisher` and `@Gateway` JavaDocs

### PR DESCRIPTION
**Pull request by artembilan**

### CODE REVIEW

File: spring-integration-core/src/main/java/org/springframework/integration/annotation/Publisher.java [hunk 1]
```diff
@@ -23,11 +23,21 @@
 
 /**
  * Annotation to indicate that a method, or all public methods if applied at
- * class-level, should publish Messages. The  @Payload and @Header annotations
- * can be used in conjunction with this to determine the content of the Message.
+ * class-level, should publish Messages.
+ * <p>
+ * The Message will be constructed from the return value of the method invocation and sent
```
Comment (garyrussell) [inline spring-integration-core/src/main/java/org/springframework/integration/annotation/Publisher.java]: By default,

**Comment by artembilan** [comment]
Pushed.

**Comment by artembilan** [comment]
Pushed more thoughts in the JavaDocs.

**Comment by garyrussell** [comment]
Polishing: https://github.com/garyrussell/spring-integration/commit/d5661f332c160daefa1ef77ae6bec9e59cbc1724

**Comment by artembilan** [comment]
Thank you Gary for the proper expressing those sentences!
Feel free to merge. Or let me know and I'll utilize your polishing on merger myself.

**Comment by garyrussell** [comment]
Merged as 1bd955bef0649bda2d6514bf293d92cb8f5a5183

**Pull request closed without merging**